torch/_inductor/runtime/triton_helpers.py

```diff
@@ -627,3 +631,42 @@ def x_grid_barrier(sem):
 
     # TODO(jansel): is this needed?
     tl.debug_barrier()
+
+
+def triton_builtin(f: _T) -> _T:
+    """
+    Decorator to mark a function as a Triton built-in function.  These functions
+    are evaluated at compile time.
+
+    Args:
+        f (function): The function to be marked as a Triton built-in.
+
+    Returns:
+        function: The same function, marked as a Triton built-in.
+    """
+    f.__triton_builtin__ = True  # type: ignore[attr-defined]
+    return f
+
+
+@triton_builtin
+def constexpr_next_power_of_2(
+    n: tl.constexpr, *, _builder: object = None
+) -> tl.constexpr:
+    """
+    A version triton.next_power_of_two that can be used within a kernel on constants.
+    """
+    assert isinstance(n, tl.constexpr)
+    return tl.constexpr(triton.next_power_of_2(n.value))
+
+
+@triton_builtin
+def zero_other(ptr: Any, mask: Any, *, _builder: object = None) -> tl.constexpr:
```

Comment:
```suggestion
def zero_other(ptr: Any, mask: Any, *, _builder: Optional[object] = None) -> tl.constexpr:
```
You probably still want to have value optional checks

Reply:
None is an object:
```
>>> isinstance(None, object)
True
```